compiler/rustc_codegen_ssa/src/back/write.rs

```diff
@@ -1421,45 +1414,30 @@ fn start_executing_work<B: ExtraBackendMethods>(
                     assert!(!started_lto);
                     started_lto = true;
 
-                    let needs_fat_lto = mem::take(&mut needs_fat_lto);
-                    let needs_thin_lto = mem::take(&mut needs_thin_lto);
-                    let import_only_modules = mem::take(&mut lto_import_only_modules);
-                    let each_linked_rlib_file_for_lto =
-                        mem::take(&mut each_linked_rlib_file_for_lto);
+                    if !needs_fat_lto.is_empty() {
```

Comment:
I think we effectively lost the call to `check_lto_allowed` on line 1422 below right? We bail out early here for fat LTO but in the new section below the loop, there's no call to `check_lto_allowed`.

Reply:
Oh sorry, I was looking for it in the wrong place. I see it in `do_fat_lto` now.